container-chains/nodes/frontier/src/rpc/finality.rs

```diff
@@ -0,0 +1,73 @@
+use fc_rpc::frontier_backend_client::{self, is_canon};
+use jsonrpsee::{core::RpcResult, proc_macros::rpc};
+use sp_blockchain::HeaderBackend;
+use sp_core::H256;
+use sp_runtime::traits::Block;
+use std::{marker::PhantomData, sync::Arc};
+
+#[rpc(server)]
+#[async_trait::async_trait]
+pub trait FrontierFinalityApi {
+    /// Reports whether a Substrate or Ethereum block is finalized.
+    /// Returns false if the block is not found.
+    #[method(name = "frnt_isBlockFinalized")]
+    fn is_block_finalized(&self, block_hash: H256) -> RpcResult<bool>;
```

Comment:
Why are we not adding the tx finalized method?

       fn is_tx_finalized(&self, tx_hash: H256) -> RpcResult<bool>;

Reply:
yes let's add this, this is fairly important